streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -4017,6 +4017,60 @@ class ZStream[-R, +E, +A](val channel: ZChannel[R, Any, Any, Any, E, Chunk[A], A
   def whenZIO[R1 <: R, E1 >: E](b: ZIO[R1, E1, Boolean])(implicit trace: ZTraceElement): ZStream[R1, E1, A] =
     ZStream.whenZIO(b)(self)
 
+  /**
+   * Emits a sliding window of n elements.
+   */
+  def windowed(chunkSize: Int)(implicit trace: ZTraceElement): ZStream[R, E, Chunk[A]] =
+    if (chunkSize <= 0) ZStream.empty
```

Comment:
I think I would die in this case and treat it as an illegal argument exception instead of returning an empty stream. That would be consistent with the behavior of `grouped(0)` on a collection and the implicit contract that this should return the same stream, just with elements grouped differently.

Reply:
Done